pkg/katatestutils/README.md

```diff
@@ -0,0 +1,175 @@
+
+Note that you could add the TestConstraint` object to the `Skip()` call as it
+will provide details of why the skip occurred:
+
+```go
+if tc.NotValid(...) {
+    t.Skipf("skipping test as requirements not met: %v", tc)
+}
+```
+
+#### Associating an issue with a constraint
+
+You can add a constraint which specifies an issue URL for the skip. No
+checking is performed on the issue but if specified, it will be added to the
+`TestConstraint` and recorded in error messages and when that object is
+displayed:
+
+```go
+if tc.NotValid(WithIssue("https://github.com/kata-containers/runtime/issues/1586"), ...) {
+    t.Skipf("skipping test as requirements not met: %v", tc)
+}
+```
+
+### Examples
+
+#### Skip tests based on user
+
+Use the `NeedRoot()` constraint to skip a test unless running as `root`:
+
+```go
+func TestOnlyRunWhenRoot(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedRoot()) {
+        t.Skip("skipping test as not running as root user")
+    }
+
+    // Test code to run as root user ...
+}
+```
+
+Use the `NeedNonRoot()` constraint to skip a test unless running as a
+non-`root` user:
+
+```go
+func TestOnlyRunWhenNotRoot(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedNonRoot()) {
+        t.Skip("skipping test as running as root user")
+    }
+
+    // Test code to run as non-root user ...
+}
+```
+
+#### Skip tests based on distro
+
+Use the `NeedDistro()` constraint to skip a test unless running on a
+particular Linux distribution:
+
+```go
+func TestOnlyRunOnUbuntu(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedDistro("ubuntu")) {
+        t.Skip("skipping test as not running on ubuntu")
+    }
+
+    // Test code to run on Ubuntu only ...
+}
+```
+
+Use the `NeedDistroNotEquals()` constraint to skip a test unless running
+on a Linux distribution other than the one specified:
+
+```go
+func TestDontRunOnFedora(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedDistroNotEquals("fedora")) {
+        t.Skip("skipping test as running on fedora")
+    }
+
+    // Test code to run on any distro apart from Fedora ...
+}
+```
+
+#### Skip tests based on kernel version
+
+Use the `NeedKernelVersionGE()` constraint to skip a test unless running on a
+system with atleast the specified kernel version:
+
+```go
+func TestNewKernelVersion(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedKernelVersionGE("5.0.10")) {
+        t.Skip("skipping test as kernel is too old")
+    }
+
+    // Test code to run on specified kernel version (or newer) ...
+}
+```
+
+Use the `NeedKernelVersionLT()` constraint to skip a test unless running on a
+system whose kernel is older than the specified kernel version:
+
+```go
+func TestOldKernelVersion(t *testing.T) {
+
+    if tc.NotValid(ktu.NeedKernelVersionLT("4.14.114")) {
+        t.Skip("skipping test as kernel is too new")
+    }
+
+    // Test code to run on specified kernel version (or newer) ...
+}
+```
+
+### Full details
+
+The public API is shown in [`constraints_api.go`](constraints_api.go).
```

Comment:
any chance we can get that as a godoc, so we can see the go api in the same style one does for other go packages?

Reply:
Updated to point at the godoc site.